### PR TITLE
Add TinyAD

### PR DESCRIPTION
**Pull request by jjcasmar**
### Summary
Changes to recipe:  **tinyad/cci.20240718**

#### Motivation
TinyAD is a library for automatic differentiation, presented at Eurographics Symposium on Geometry Processing 2022

#### Details
Add the library

---
- [X ] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [X ] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [X ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by valgur** [comment]
The project also relies on OpenMP and warns when it is unavailable. The package would really benefit from #22360. For now you will have to export the appropriate compiler and linker flags for each separate compiler to make sure it's enabled and linked against by the consuming projects.

File: recipes/tinyad/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "TinyAD")
+        self.cpp_info.set_property("cmake_target_name", "TinyAD")
+        self.cpp_info.components["tinyad"].set_property("cmake_target_name", "TinyAD::TinyAD")
+
+        self.cpp_info.components["tinyad"].bindirs = []
+        self.cpp_info.components["tinyad"].libdirs = []
+        self.cpp_info.components["tinyad"].includedirs = [os.path.join(self.package_folder, "include")]
+        self.cpp_info.components["tinyad"].requires = ["eigen::eigen"]
+
```
[the remark this answers is not in the document]
Comment (jjcasmar) [reply recipes/tinyad/all/conanfile.py:53]: The cmake_target_name should be TinyAD alone, as that is the one used in the upstream CMake.

[the remark this answers is not in the document]
Comment (valgur) [reply recipes/tinyad/all/conanfile.py:53]: Correct. I adjusted my adjusted change accordingly.

**Comment by jjcasmar** [comment]
Is the CI nor working?

**Comment by AbrilRBS** [comment]
@jjcasmar we're under maintenance, the service should be available again soon, thanks for your patience :)

**Comment by jjcasmar** [comment]
Any update on this?

**Review by jcar87 — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]
Finally!

**Pull request merged**